vagrant/spec/README.md

```diff
@@ -0,0 +1,93 @@
+# cluster.k8s.io/v1alpha1
+
+## kind: Cluster
+
+Allows to specify some characteristics of the cluster, selecting among all the supported kubeadm options. You can define:
+
+      - The target kubernetes/kubeadm version
+      - How the controlplane will be deployed (static pods vs self hosting)
+      - Which type of certificate authority are you going to use (local vs external)
+      - Where your PKI will be stored (filesystem or secrets)
+      - Which type of DNS are you going to use (kubeDNS or coreDNS)
+      - Which type of pod network add-on are you going to use (weavenet, flannel, calico)
+      - Which type of kubelet config are you going to use (dystemd dropIn files, dynamic Kubelet config)
+      - TODO: Which type of network are you using (Ipv4, Ipv6)
+      - TODO: Which type of environment are you simulating (with or without internet connection) 
+      - TODO: Auditing  
+      - ...
+
+Most of the configuration should be done in the providerConfig object
+
+``` yaml
+apiVersion: cluster.k8s.io/v1alpha1
+kind: Cluster
+metadata:
+  name: kubeadm-test
+spec:
+  providerConfig: 
+    value: 
+      apiVersion: vagrantCluster/v1alpha1
+      kind: vagrantCluster
+      kubernetes:
+        version:
+          <string (optional, default 1.10.2)
+          defines the version of kubeadm, kubelet, kubectl binaries and of the version of the controlplane components to be installed>
+        controlPlaneType:
+          <staticPods|selfHosting (optional, default staticPods)
+          defines how the controlplane will be deployed>
+        certificateAuthority:
+          <local|external (optional, default local)
+          defines the type of certificate authority that will be used. local certificateAuthority are created by kubeadm, while external certificateAuthority should be provided before creating the cluster>
+        pkiLocation:
+          <filesystem|secrets (optional, default filesystem)
+          defines the target location for the certificate authority>
+        dnsAddon:
+          <kubeDNS|coreDNS (optional, default kubeDNS)
+          defines the target DNS addon of choice>
+        kubeletConfig:
+          <systemdDropIn|dynamicKubeletConfig (optional, default systemdDropIn)
+          defines the target DNS addon of choice>
+        networkAddon:
+          <weavenet|flannel|calico (optional, default weavenet)
+          defines the target network add-on of choice>
+      etcd:
+        version: <string (optional, default v3.2.17), defines the version of etcd binaries to be installed - only extenal etcd - >
+```
```

Comment:
I really like how you're using Cluster API for this.
It allows us to maintain a similar experience if we recreate this workflow for other tools such as kubeadm-dind-cluster.

It might also be possible to use other Cluster API controllers with this dev-tooling as well.

Reply:
totally agree. 
Now the ansible part and the vagrant part are even more decoupled, and it will be easier to introduce e.g. controllers or other tools like terraform as alternative to vagrant